x-pack/plugins/observability_ai_assistant/server/service/util/flush_buffer.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { repeat } from 'lodash';
+import { Observable, OperatorFunction } from 'rxjs';
+import {
+  BufferFlushEvent,
+  StreamingChatResponseEventType,
+  StreamingChatResponseEventWithoutError,
+} from '../../../common/conversation_complete';
+
+export function flushBuffer<T extends StreamingChatResponseEventWithoutError>(
```

Comment:
Why do we need this in the first place? Is it because the cloud proxy requires it? It would be helpful with the background if this is indeed necessitated by Elastic Cloud infra needs.

Reply:
Yes - will add a comment.